lib/http-context.js

```diff
@@ -184,7 +190,7 @@ function coerce(str) {
   if ('true' === str) return true;
   if ('false' === str) return false;
   if (isfloat.test(str)) return parseFloat(str, 10);
-  if (isint.test(str)) return parseInt(str, 10);
+  if (isint.test(str) && str.charAt(0) !== '0') return parseInt(str, 10);
```

Comment:
IIUC, this change is not covered by unit-tests, as I don't see any test for an int-value starting with a zero, e.g. "0123".